### ISSUE TITLE
Global ComputeTargetHTTPSProxy: failed to fetch live state

### ISSUE THREAD

**Issue by torfjor** [GoogleCloudPlatform/k8s-config-connector#1841]
### Checklist

- [X] I did not find a related open issue.
- [X] I did not find a solution in the troubleshooting guide: (https://cloud.google.com/config-connector/docs/troubleshooting)
- [X] If this issue is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

### Bug Description

Desired resource:
```yaml
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeTargetHTTPSProxy
metadata:
  annotations:
    cnrm.cloud.google.com/deletion-policy: abandon
    cnrm.cloud.google.com/project-id: pgs-project-prod
    cnrm.cloud.google.com/state-into-spec: merge
  name: trust
  namespace: spire-federation
spec:
  certificateMapRef:
    name: trust
  location: global
  urlMapRef:
    name: trust
```

The target resource is created, but reading the live state fails:
```shell
$ kubectl logs -n cnrm-system -l cnrm.cloud.google.com/scoped-namespace=spire-federation | grep computetargethttpsproxy-controller | jq
{
  "severity": "info",
  "timestamp": "2024-05-18T06:46:39.605Z",
  "logger": "computetargethttpsproxy-controller",
  "msg": "starting reconcile",
  "resource": {
    "name": "trust",
    "namespace": "spire-federation"
  }
}
{
  "severity": "error",
  "timestamp": "2024-05-18T06:46:39.648Z",
  "msg": "Reconciler error",
  "controller": "computetargethttpsproxy-controller",
  "controllerGroup": "compute.cnrm.cloud.google.com",
  "controllerKind": "ComputeTargetHTTPSProxy",
  "ComputeTargetHTTPSProxy": {
    "name": "trust",
    "namespace": "spire-federation"
  },
  "namespace": "spire-federation",
  "name": "trust",
  "reconcileID": "d9471c39-725f-47bf-a3f5-dbe51f55867c",
  "error": "Update call failed: error fetching live state: error converting resource config: unable to resolve missing value: location"
}
```

Glancing at the code it looks like some global resources get their `"location": "global"` fields removed when translated to the underlying API. The raw response returned from the computev1 API for this target https proxy doesn't include a `location` field.

### Additional Diagnostic Information

None

### Kubernetes Cluster Version

v1.27.5-gke.200

### Config Connector Version

1.117.0

### Config Connector Mode

namespaced mode (default)

### Steps to reproduce the issue

Apply yaml.

**Comment by ed-marks** [comment]
ran into this recently - looks like it was fixed in 1.120.0

**Comment by torfjor** [comment]
Closing as resolved in 1.120.1
